zerver/openapi/zulip.yaml

```diff
@@ -17079,6 +17164,64 @@ components:
         - to
         - topic
         - content
+    ScheduledMessage:
+      type: object
+      description: |
+        A dictionary for representing a scheduled message.
+      properties:
+        message_id:
+          type: integer
+          description: |
+            The unique ID of the scheduled message. It can be used to modify and
+            delete the scheduled message.
+        type:
+          type: string
+          description: |
+            The type of the scheduled message. Either unaddressed (empty string), "stream",
+            or "private" (for PMs and private group messages).
+          enum:
+            - ""
+            - stream
+            - private
+        to:
+          type: array
+          description: |
+            An array of the tentative target audience IDs. For "stream"
+            messages, this should contain exactly 1 ID, the ID of the
+            target stream. For private messages, this should be an array
+            of target user IDs. This cannot be an empty array since
+            scheduled messages are always addressed.
+          items:
+            type: integer
+        topic:
+          type: string
+          description: |
+            For stream scheduled message, the tentative topic name. For private messages,
+            this will be ignored and should ideally
+            be the empty string. Should not contain null bytes.
```

Comment:
This text (with "ignored" and "should") seems like it's more aimed at describing parameters to create a scheduled message than at describing what the server returns when asked about them (whereas the latter is what this bit of spec is actually used for at present, IIUC.)

In that context of creating a scheduled message, what I'd want to do from the client side is not to send an empty string here, but rather to not send this parameter at all. Conversely, I'd ideally want the server to not include this property at all when it's a DM, rather than include it as the empty string.

Reply:
Agreed, this also looks like it was copied from Drafts and could be greatly simplified.